In [88]:
#import cartopy.crs as ccrs
#import cartopy.feature as cfeature
import obspy
import matplotlib.patches as mpatches
import matplotlib.pyplot as plt
import numpy as np
from obspy import read
from obspy import UTCDateTime
from obspy.clients.fdsn import Client
from obspy.taup import plot_travel_times, TauPyModel
import pandas as pd 
from obspy.clients.fdsn.header import URL_MAPPINGS
#import pyproj

In [89]:

df_test = pd.read_csv("csvs/earthquake_metadata.csv")
#df_test = pd.read_csv("csvs/chunk5.csv")

df_test.set_index('trace_name',inplace= True)

C:\ProgramData\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3165: DtypeWarning: Columns (7,11,13,14,15,18,19,20,21,22,24,25,26,30,31) have mixed types.Specify dtype option on import or set low_memory=False.
  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,


In [90]:
df_test.shape

(1265657, 34)

In [91]:
df_test['trace_start_time'] = pd.to_datetime(df_test['trace_start_time'],format="%Y-%m-%dT%H:%M:%S.%f")


In [92]:
df_test['station'] = df_test['network_code'] + '.' + df_test['receiver_code']

In [93]:
df_test['station']

trace_name
109C.TA_201510210555_NO     TA.109C
109C.TA_201511061450_NO     TA.109C
109C.TA_201511070220_NO     TA.109C
109C.TA_201511140515_NO     TA.109C
109C.TA_201512251850_NO     TA.109C
                             ...   
YHL.WY_20170621094042_EV     WY.YHL
YHL.WY_20170621122104_EV     WY.YHL
YHL.WY_20170621122339_EV     WY.YHL
YHL.WY_20170621133439_EV     WY.YHL
YHL.WY_20170621140610_EV     WY.YHL
Name: station, Length: 1265657, dtype: object

In [94]:
earliest_trace  = df_test[df_test['trace_start_time'] == df_test['trace_start_time'].min()]

In [95]:
df_test.head(1)

network_code receiver_code receiver_type  \
trace_name                                                         
109C.TA_201510210555_NO           TA          109C            HH   

                         receiver_latitude  receiver_longitude  \
trace_name                                                       
109C.TA_201510210555_NO            32.8889           -117.1051   

                         receiver_elevation_m  p_arrival_sample p_status  \
trace_name                                                                 
109C.TA_201510210555_NO                 150.0               NaN      NaN   

                         p_weight  p_travel_sec  ...  source_magnitude_author  \
trace_name                                       ...                            
109C.TA_201510210555_NO       NaN           NaN  ...                      NaN   

                        source_mechanism_strike_dip_rake  source_distance_deg  \
trace_name                                                                      
109C.TA_201510210555_NO                              NaN                  NaN   

                        source_distance_km back_azimuth_deg snr_db  \
trace_name                                                           
109C.TA_201510210555_NO                NaN              NaN    NaN   

                         coda_end_sample    trace_start_time trace_category  \
trace_name                                                                    
109C.TA_201510210555_NO              NaN 2015-10-21 05:55:00          noise   

                         station  
trace_name                        
109C.TA_201510210555_NO  TA.109C  

[1 rows x 35 columns]


for key in sorted(URL_MAPPINGS.keys()):
    client = Client(key)
#for key,client in clients_dict.items():
    try:
        station_test = client.get_stations(latitude=32.8889	, longitude=-117.1051)
        print(key)
    except:
        pass


In [96]:
earliest_trace

network_code receiver_code receiver_type  \
trace_name                                                          
CAL.NC_19840424221806_EV           NC           CAL            EL   

                          receiver_latitude  receiver_longitude  \
trace_name                                                        
CAL.NC_19840424221806_EV          37.450348         -121.800369   

                          receiver_elevation_m  p_arrival_sample p_status  \
trace_name                                                                  
CAL.NC_19840424221806_EV                 251.0             600.0   manual   

                          p_weight  p_travel_sec  ...  \
trace_name                                        ...   
CAL.NC_19840424221806_EV      0.51          4.64  ...   

                          source_magnitude_author  \
trace_name                                          
CAL.NC_19840424221806_EV                       NC   

                         source_mechanism_strike_dip_rake  \
trace_name                                                  
CAL.NC_19840424221806_EV                             None   

                          source_distance_deg source_distance_km  \
trace_name                                                         
CAL.NC_19840424221806_EV               0.2117              23.37   

                         back_azimuth_deg                              snr_db  \
trace_name                                                                      
CAL.NC_19840424221806_EV            147.0  [4.5        1.79999995 4.9000001 ]   

                          coda_end_sample        trace_start_time  \
trace_name                                                          
CAL.NC_19840424221806_EV        [[1900.]] 1984-04-24 22:18:07.960   

                            trace_category station  
trace_name                                          
CAL.NC_19840424221806_EV  earthquake_local  NC.CAL  

[1 rows x 35 columns]

In [97]:
#client = Client("IRIS")
client = Client("IRIS")
first_event = UTCDateTime(df_test['trace_start_time'].min().isoformat())
last_event = UTCDateTime(df_test['trace_start_time'].max().isoformat())



iris_events = client.get_events(starttime  = first_event,
endtime=last_event,
eventtype = "earthquake")
inventory = client.get_stations(latitude=37.34164, longitude=-121.64257	)
print(inventory)  

In [98]:
df_test_closest = df_test[df_test['source_distance_km'] == min(df_test['source_distance_km'])]

In [99]:
df_test_closest.dtypes

network_code                                object
receiver_code                               object
receiver_type                               object
receiver_latitude                          float64
receiver_longitude                         float64
receiver_elevation_m                       float64
p_arrival_sample                           float64
p_status                                    object
p_weight                                   float64
p_travel_sec                               float64
s_arrival_sample                           float64
s_status                                    object
s_weight                                   float64
source_id                                   object
source_origin_time                          object
source_origin_uncertainty_sec               object
source_latitude                            float64
source_longitude                           float64
source_error_sec                            object
source_gap_deg                 

In [100]:
from obspy import read_inventory, UTCDateTime
inv = read_inventory()

stations = client.get_stations(starttime  = first_event,
endtime=last_event,level='station')
#.get_contents()["stations"]
#["stations"]
#.get_contents()["stations"]

for key in sorted(URL_MAPPINGS.keys()):
    print(key)

#for s in stations.get_contents()["stations"]:
stations_per_provider = {}
providers = list(URL_MAPPINGS.keys())
faulty_providers = []
#providers.remove('BGR')
for key in providers:
    #client = Client(key)
    try:
        client = Client(key)
        stations_per_provider[key] = []
#client = Client("IRIS")
    
        stations = client.get_stations(starttime  = first_event,
        endtime=last_event,level='station')
        for s in stations:
            for e in s:
            #print()
                stations_per_provider[key].append(s.code + '.' + e.code)
    except:
        #print(key)
        faulty_providers.append(key)
        pass
        #raise ValueError
    #station_coords = {}
    #for s in stations:
    #    for e in s:
            #print()
    #        stations_per_provider[key].append(s.code + '.' + e.code)
            #station_coords[s.code + '.' + e.code] = (e.latitude,e.longitude)
            #print(s.code + '.' + e.code)
            #print(e.code)

    

stations_per_provider

        client = Client(key)
        stations_per_provider[key] = []
#client = Client("IRIS")
    
        stations = client.get_stations(starttime  = first_event,
        endtime=last_event,level='station')

In [190]:
metadata_stations = set(df_test['station'])
net_reciever_pairs = []
for m in metadata_stations:
    #print(m)
    #try:
    if  isinstance(m,str):
        m_split = m.split('.')
        #print(m_split)
        net_reciever_pairs.append((m_split[0],m_split[1],"*", "*",first_event, last_event))
    #except:
    #    print(m)
    #    print(type(m))
    #    pass

In [152]:
df_test

network_code receiver_code receiver_type  \
trace_name                                                          
109C.TA_201510210555_NO            TA          109C            HH   
109C.TA_201511061450_NO            TA          109C            HH   
109C.TA_201511070220_NO            TA          109C            HH   
109C.TA_201511140515_NO            TA          109C            HH   
109C.TA_201512251850_NO            TA          109C            HH   
...                               ...           ...           ...   
YHL.WY_20170621094042_EV           WY           YHL            HH   
YHL.WY_20170621122104_EV           WY           YHL            HH   
YHL.WY_20170621122339_EV           WY           YHL            HH   
YHL.WY_20170621133439_EV           WY           YHL            HH   
YHL.WY_20170621140610_EV           WY           YHL            HH   

                          receiver_latitude  receiver_longitude  \
trace_name                                                        
109C.TA_201510210555_NO             32.8889           -117.1051   
109C.TA_201511061450_NO             32.8889           -117.1051   
109C.TA_201511070220_NO             32.8889           -117.1051   
109C.TA_201511140515_NO             32.8889           -117.1051   
109C.TA_201512251850_NO             32.8889           -117.1051   
...                                     ...                 ...   
YHL.WY_20170621094042_EV            44.8509           -111.1830   
YHL.WY_20170621122104_EV            44.8509           -111.1830   
YHL.WY_20170621122339_EV            44.8509           -111.1830   
YHL.WY_20170621133439_EV            44.8509           -111.1830   
YHL.WY_20170621140610_EV            44.8509           -111.1830   

                          receiver_elevation_m  p_arrival_sample p_status  \
trace_name                                                                  
109C.TA_201510210555_NO                  150.0               NaN      NaN   
109C.TA_201511061450_NO                  150.0               NaN      NaN   
109C.TA_201511070220_NO                  150.0               NaN      NaN   
109C.TA_201511140515_NO                  150.0               NaN      NaN   
109C.TA_201512251850_NO                  150.0               NaN      NaN   
...                                        ...               ...      ...   
YHL.WY_20170621094042_EV                2691.0             400.0   manual   
YHL.WY_20170621122104_EV                2691.0             600.0   manual   
YHL.WY_20170621122339_EV                2691.0             500.0   manual   
YHL.WY_20170621133439_EV                2691.0             500.0   manual   
YHL.WY_20170621140610_EV                2691.0             500.0   manual   

                          p_weight  p_travel_sec  ...  \
trace_name                                        ...   
109C.TA_201510210555_NO        NaN           NaN  ...   
109C.TA_201511061450_NO        NaN           NaN  ...   
109C.TA_201511070220_NO        NaN           NaN  ...   
109C.TA_201511140515_NO        NaN           NaN  ...   
109C.TA_201512251850_NO        NaN           NaN  ...   
...                            ...           ...  ...   
YHL.WY_20170621094042_EV      0.68          4.21  ...   
YHL.WY_20170621122104_EV      0.65          4.84  ...   
YHL.WY_20170621122339_EV      0.65          4.82  ...   
YHL.WY_20170621133439_EV      0.59          4.26  ...   
YHL.WY_20170621140610_EV      0.59          4.45  ...   

                          source_magnitude_author  \
trace_name                                          
109C.TA_201510210555_NO                       NaN   
109C.TA_201511061450_NO                       NaN   
109C.TA_201511070220_NO                       NaN   
109C.TA_201511140515_NO                       NaN   
109C.TA_201512251850_NO                       NaN   
...                                           ...   
YHL.WY_20170621094042_EV                     None   
YHL.WY_20170621122104_EV          

Stations file creation

In [191]:
stations_df = pd.DataFrame(columns=['provider', 'network_name','station_name', 'latitude','longitude','station_opened','station_closed'])

In [192]:
stations_per_provider = {}
providers = list(URL_MAPPINGS.keys())
faulty_providers = []
#providers.remove('BGR')
#print(providers)
for key in providers:
    #client = Client(key)
    try:
        client = Client(key)
        stations_per_provider[key] = []
    #client = Client("IRIS")

        #stations = client.get_stations(starttime  = first_event,
        #endtime=last_event,level='station')
        stations = client.get_stations_bulk(net_reciever_pairs,level='station')
        for s in stations:
            for e in s:
                row_frame = pd.Series([key,s.code,e.code,e.latitude,e.longitude,e.creation_date,e.termination_date]).to_frame().T                
                #row_frame = station_row.to_frame().T
                row_frame.columns = stations_df.columns

                stations_df = pd.concat([stations_df,row_frame],names = list(stations_df.columns), join = 'outer')
                #print(station_row)
                #stations_df = pd.concat([stations_df,station_row],ignore_index = False, axis = 0)
                #print(stations_df)
                
            #print()
                #stations_per_provider[key].append(s.code + '.' + e.code)
    except:
        #print(key)
        faulty_providers.append(key)
        pass

C:\ProgramData\Anaconda3\lib\site-packages\obspy\io\stationxml\core.py:96: UserWarning: The StationXML file has version 1, ObsPy can read versions (1.0, 1.1). Proceed with caution.
  warnings.warn("The StationXML file has version %s, ObsPy can "


In [195]:
stations_df.to_csv('station_definitions_from_metadata.csv')

In [143]:
#stations_df.to_csv('stations_demo.csv', index = False)

In [199]:
df_test

network_code receiver_code receiver_type  \
trace_name                                                          
109C.TA_201510210555_NO            TA          109C            HH   
109C.TA_201511061450_NO            TA          109C            HH   
109C.TA_201511070220_NO            TA          109C            HH   
109C.TA_201511140515_NO            TA          109C            HH   
109C.TA_201512251850_NO            TA          109C            HH   
...                               ...           ...           ...   
YHL.WY_20170621094042_EV           WY           YHL            HH   
YHL.WY_20170621122104_EV           WY           YHL            HH   
YHL.WY_20170621122339_EV           WY           YHL            HH   
YHL.WY_20170621133439_EV           WY           YHL            HH   
YHL.WY_20170621140610_EV           WY           YHL            HH   

                          receiver_latitude  receiver_longitude  \
trace_name                                                        
109C.TA_201510210555_NO             32.8889           -117.1051   
109C.TA_201511061450_NO             32.8889           -117.1051   
109C.TA_201511070220_NO             32.8889           -117.1051   
109C.TA_201511140515_NO             32.8889           -117.1051   
109C.TA_201512251850_NO             32.8889           -117.1051   
...                                     ...                 ...   
YHL.WY_20170621094042_EV            44.8509           -111.1830   
YHL.WY_20170621122104_EV            44.8509           -111.1830   
YHL.WY_20170621122339_EV            44.8509           -111.1830   
YHL.WY_20170621133439_EV            44.8509           -111.1830   
YHL.WY_20170621140610_EV            44.8509           -111.1830   

                          receiver_elevation_m  p_arrival_sample p_status  \
trace_name                                                                  
109C.TA_201510210555_NO                  150.0               NaN      NaN   
109C.TA_201511061450_NO                  150.0               NaN      NaN   
109C.TA_201511070220_NO                  150.0               NaN      NaN   
109C.TA_201511140515_NO                  150.0               NaN      NaN   
109C.TA_201512251850_NO                  150.0               NaN      NaN   
...                                        ...               ...      ...   
YHL.WY_20170621094042_EV                2691.0             400.0   manual   
YHL.WY_20170621122104_EV                2691.0             600.0   manual   
YHL.WY_20170621122339_EV                2691.0             500.0   manual   
YHL.WY_20170621133439_EV                2691.0             500.0   manual   
YHL.WY_20170621140610_EV                2691.0             500.0   manual   

                          p_weight  p_travel_sec  ...  \
trace_name                                        ...   
109C.TA_201510210555_NO        NaN           NaN  ...   
109C.TA_201511061450_NO        NaN           NaN  ...   
109C.TA_201511070220_NO        NaN           NaN  ...   
109C.TA_201511140515_NO        NaN           NaN  ...   
109C.TA_201512251850_NO        NaN           NaN  ...   
...                            ...           ...  ...   
YHL.WY_20170621094042_EV      0.68          4.21  ...   
YHL.WY_20170621122104_EV      0.65          4.84  ...   
YHL.WY_20170621122339_EV      0.65          4.82  ...   
YHL.WY_20170621133439_EV      0.59          4.26  ...   
YHL.WY_20170621140610_EV      0.59          4.45  ...   

                          source_magnitude_author  \
trace_name                                          
109C.TA_201510210555_NO                       NaN   
109C.TA_201511061450_NO                       NaN   
109C.TA_201511070220_NO                       NaN   
109C.TA_201511140515_NO                       NaN   
109C.TA_201512251850_NO                       NaN   
...                                           ...   
YHL.WY_20170621094042_EV                     None   
YHL.WY_20170621122104_EV          

In [206]:
df_test['station_lon_lat'] = df_test['receiver_longitude'].astype(str) + ',' + df_test['receiver_latitude'].astype(str)

In [207]:
df_test['station_lon_lat']

trace_name
109C.TA_201510210555_NO     -117.1051,32.8889
109C.TA_201511061450_NO     -117.1051,32.8889
109C.TA_201511070220_NO     -117.1051,32.8889
109C.TA_201511140515_NO     -117.1051,32.8889
109C.TA_201512251850_NO     -117.1051,32.8889
                                  ...        
YHL.WY_20170621094042_EV     -111.183,44.8509
YHL.WY_20170621122104_EV     -111.183,44.8509
YHL.WY_20170621122339_EV     -111.183,44.8509
YHL.WY_20170621133439_EV     -111.183,44.8509
YHL.WY_20170621140610_EV     -111.183,44.8509
Name: station_lon_lat, Length: 1265657, dtype: object

In [202]:
df_test['station_lon_lat'].head(2)

trace_name
109C.TA_201510210555_NO    trace_name\n109C.TA_201510210555_NO    -117.10...
109C.TA_201511061450_NO    trace_name\n109C.TA_201510210555_NO    -117.10...
Name: station_lon_lat, dtype: object

In [154]:
#stations_df.set_index('provider')
stations_df['station'] = stations_df['network_name'] + '.' + stations_df['station_name']

In [208]:
stations_df['lon_lat'] = stations_df['longitude'].astype(str) + ',' + stations_df['latitude'].astype(str)

In [220]:
df_test_res= df_test.reset_index()
some_check_df = stations_df.merge(df_test_res,how='inner',left_on = 'lon_lat', right_on = 'station_lon_lat')

In [223]:
some_check_df['trace_name']

0           CLZ.GR_19960820124028_EV
1           CLZ.GR_19961023021810_EV
2           CLZ.GR_19970207112804_EV
3           CLZ.GR_19970218083051_EV
4           CLZ.GR_20010225180052_EV
                     ...            
716202    MAN01.ON_20141014200133_EV
716203    MAN01.ON_20141028133352_EV
716204    MAN01.ON_20150509145842_EV
716205    MAN01.ON_20150525013951_EV
716206    VAS01.ON_20141014200138_EV
Name: trace_name, Length: 716207, dtype: object

In [241]:
na_drop = df_test.dropna(subset=['source_longitude','source_latitude'])

network_code receiver_code receiver_type  \
trace_name                                                           
109C.TA_20060723155859_EV           TA          109C            BH   
109C.TA_20061103155652_EV           TA          109C            BH   
109C.TA_20061103161223_EV           TA          109C            BH   
109C.TA_20061114133221_EV           TA          109C            BH   
109C.TA_20061127104640_EV           TA          109C            BH   
...                                ...           ...           ...   
YHL.WY_20170621094042_EV            WY           YHL            HH   
YHL.WY_20170621122104_EV            WY           YHL            HH   
YHL.WY_20170621122339_EV            WY           YHL            HH   
YHL.WY_20170621133439_EV            WY           YHL            HH   
YHL.WY_20170621140610_EV            WY           YHL            HH   

                           receiver_latitude  receiver_longitude  \
trace_name                                                         
109C.TA_20060723155859_EV            32.8889           -117.1051   
109C.TA_20061103155652_EV            32.8889           -117.1051   
109C.TA_20061103161223_EV            32.8889           -117.1051   
109C.TA_20061114133221_EV            32.8889           -117.1051   
109C.TA_20061127104640_EV            32.8889           -117.1051   
...                                      ...                 ...   
YHL.WY_20170621094042_EV             44.8509           -111.1830   
YHL.WY_20170621122104_EV             44.8509           -111.1830   
YHL.WY_20170621122339_EV             44.8509           -111.1830   
YHL.WY_20170621133439_EV             44.8509           -111.1830   
YHL.WY_20170621140610_EV             44.8509           -111.1830   

                           receiver_elevation_m  p_arrival_sample p_status  \
trace_name                                                                   
109C.TA_20060723155859_EV                 150.0             700.0   manual   
109C.TA_20061103155652_EV                 150.0             600.0   manual   
109C.TA_20061103161223_EV                 150.0             500.0   manual   
109C.TA_20061114133221_EV                 150.0             900.0   manual   
109C.TA_20061127104640_EV                 150.0             700.0   manual   
...                                         ...               ...      ...   
YHL.WY_20170621094042_EV                 2691.0             400.0   manual   
YHL.WY_20170621122104_EV                 2691.0             600.0   manual   
YHL.WY_20170621122339_EV                 2691.0             500.0   manual   
YHL.WY_20170621133439_EV                 2691.0             500.0   manual   
YHL.WY_20170621140610_EV                 2691.0             500.0   manual   

                           p_weight  p_travel_sec  ...  \
trace_name                                         ...   
109C.TA_20060723155859_EV      0.50     17.080000  ...   
109C.TA_20061103155652_EV      0.50     16.879999  ...   
109C.TA_20061103161223_EV      0.50     17.260000  ...   
109C.TA_20061114133221_EV      0.50     17.280001  ...   
109C.TA_20061127104640_EV      0.50     18.139999  ...   
...                             ...           ...  ...   
YHL.WY_20170621094042_EV       0.68      4.210000  ...   
YHL.WY_20170621122104_EV       0.65      4.840000  ...   
YHL.WY_20170621122339_EV       0.65      4.820000  ...   
YHL.WY_20170621133439_EV       0.59      4.260000  ...   
YHL.WY_20170621140610_EV       0.59      4.450000  ...   

                           source_mechanism_strike_dip_rake  \
trace_name                                                    
109C.TA_20060723155859_EV                              None   
109C.TA_20061103155652_EV                              None   
109C.TA_20061103161223_EV                              None   
109C.TA_20061114133221_EV                              None   
109C.TA_20061127104640_EV                              None   
...                

In [243]:
dup_df = na_drop[(na_drop.duplicated(['source_longitude','source_latitude','trace_start_time']))]

In [244]:
dup_df

network_code receiver_code receiver_type  \
trace_name                                                           
263.X2_20060421183717_EV            X2           263            BH   
AIND.HV_20150904001824_EV           HV          AIND            HH   
AIND.HV_20161013160957_EV           HV          AIND            HH   
AKRB.AV_20150131044634_EV           AV          AKRB            BH   
AKRB.AV_20150131044635_EV           AV          AKRB            BH   
...                                ...           ...           ...   
YER.NN_20160806070027_EV            NN           YER            HH   
YER.NN_20170102053346_EV            NN           YER            HH   
YER.NN_20170616053056_EV            NN           YER            HH   
YFT.WY_20150508093400_EV            WY           YFT            HH   
YHH.WY_20150522091746_EV            WY           YHH            HH   

                           receiver_latitude  receiver_longitude  \
trace_name                                                         
263.X2_20060421183717_EV           39.785000          112.626000   
AIND.HV_20150904001824_EV          19.372067         -155.457228   
AIND.HV_20161013160957_EV          19.372067         -155.457228   
AKRB.AV_20150131044634_EV          54.129200         -166.070800   
AKRB.AV_20150131044635_EV          54.129200         -166.070800   
...                                      ...                 ...   
YER.NN_20160806070027_EV           38.985199         -119.240601   
YER.NN_20170102053346_EV           38.985199         -119.240601   
YER.NN_20170616053056_EV           38.985199         -119.240601   
YFT.WY_20150508093400_EV           44.451330         -110.835830   
YHH.WY_20150522091746_EV           44.788330         -110.850500   

                           receiver_elevation_m  p_arrival_sample    p_status  \
trace_name                                                                      
263.X2_20060421183717_EV                 1419.0            742.96  autopicker   
AIND.HV_20150904001824_EV                1524.0            801.00  autopicker   
AIND.HV_20161013160957_EV                1524.0            900.00      manual   
AKRB.AV_20150131044634_EV                 334.0            700.00      manual   
AKRB.AV_20150131044635_EV                 334.0            600.00      manual   
...                                         ...               ...         ...   
YER.NN_20160806070027_EV                 1857.0            500.00      manual   
YER.NN_20170102053346_EV                 1857.0            600.00      manual   
YER.NN_20170616053056_EV                 1857.0            600.00      manual   
YFT.WY_20150508093400_EV                 2292.0            900.00      manual   
YHH.WY_20150522091746_EV                 2717.0            600.00      manual   

                           p_weight  p_travel_sec  ...  \
trace_name                                         ...   
263.X2_20060421183717_EV       0.92      11.70000  ...   
AIND.HV_20150904001824_EV      0.93       2.99000  ...   
AIND.HV_20161013160957_EV      0.58       4.18000  ...   
AKRB.AV_20150131044634_EV      0.50      14.62000  ...   
AKRB.AV_20150131044635_EV      0.50      14.62000  ...   
...                             ...           ...  ...   
YER.NN_20160806070027_EV       0.48      15.65523  ...   
YER.NN_20170102053346_EV       0.08      13.39680  ...   
YER.NN_20170616053056_EV       0.49      12.89372  ...   
YFT.WY_20150508093400_EV       0.63       5.18000  ...   
YHH.WY_20150522091746_EV       0.66       5.53000  ...   

                           source_mechanism_strike_dip_rake  \
trace_name                                                    
263.X2_20060421183717_EV                               None   
AIND.HV_20150904001824_EV                              None   
AIND.HV_20161013160957_EV                              None   
AKRB.AV_20150131044634_EV                              None   
AKRB.AV_20150131044635_EV                  

In [247]:
dup_df[['network_code','receiver_code','source_longitude','source_latitude','trace_start_time']].head(10)

network_code receiver_code  source_longitude  \
trace_name                                                               
263.X2_20060421183717_EV            X2           263         112.75000   
AIND.HV_20150904001824_EV           HV          AIND        -155.41450   
AIND.HV_20161013160957_EV           HV          AIND        -155.28167   
AKRB.AV_20150131044634_EV           AV          AKRB        -166.66450   
AKRB.AV_20150131044635_EV           AV          AKRB        -166.66450   
AKRB.AV_20160115072707_EV           AV          AKRB        -166.74720   
AKS.AV_20140815183015_EV            AV           AKS        -165.75470   
AKSA.AV_20140915080051_EV           AV          AKSA        -163.58600   
AKSA.AV_20150312000559_EV           AV          AKSA        -165.32660   
AKSA.AV_20160705223245_EV           AV          AKSA        -165.40690   

                           source_latitude           trace_start_time  
trace_name                                                             
263.X2_20060421183717_EV          39.17000 2006-04-21 18:37:18.070400  
AIND.HV_20150904001824_EV         19.47617 2015-09-04 00:18:25.210000  
AIND.HV_20161013160957_EV         19.37933 2016-10-13 16:09:58.840000  
AKRB.AV_20150131044634_EV         53.58590 2015-01-31 04:46:35.060000  
AKRB.AV_20150131044635_EV         53.58590 2015-01-31 04:46:36.060000  
AKRB.AV_20160115072707_EV         53.16690 2016-01-15 07:27:07.060000  
AKS.AV_20140815183015_EV          53.45380 2014-08-15 18:30:15.660000  
AKSA.AV_20140915080051_EV         54.38500 2014-09-15 08:00:51.460000  
AKSA.AV_20150312000559_EV         53.49690 2015-03-12 00:05:59.040000  
AKSA.AV_20160705223245_EV         53.12580 2016-07-05 22:32:45.920000

In [158]:
metadata_stations = set(df_test['station'])

In [174]:
metadata_stations = set(df_test['station'])


In [173]:
metadata_stations.remove(float('nan'))

KeyError: nan

In [186]:
net_reciever_pairs = []
for m in metadata_stations:
    #print(m)
    #try:
    if  isinstance(m,str):
        m_split = m.split('.')
        #print(m_split)
        net_reciever_pairs.append((m_split[0],m_split[1],"*", "*",first_event, last_event))
    #except:
    #    print(m)
    #    print(type(m))
    #    pass

In [187]:
net_reciever_pairs

[('IV',
  'T0811',
  '*',
  '*',
  1984-04-24T22:18:07.960000Z,
  2020-04-04T14:50:24.000000Z),
 ('TA',
  'X36A',
  '*',
  '*',
  1984-04-24T22:18:07.960000Z,
  2020-04-04T14:50:24.000000Z),
 ('PR',
  'LSP',
  '*',
  '*',
  1984-04-24T22:18:07.960000Z,
  2020-04-04T14:50:24.000000Z),
 ('TA',
  'O58A',
  '*',
  '*',
  1984-04-24T22:18:07.960000Z,
  2020-04-04T14:50:24.000000Z),
 ('LD',
  'FRNY',
  '*',
  '*',
  1984-04-24T22:18:07.960000Z,
  2020-04-04T14:50:24.000000Z),
 ('NM',
  'GOBM',
  '*',
  '*',
  1984-04-24T22:18:07.960000Z,
  2020-04-04T14:50:24.000000Z),
 ('US',
  'KNB',
  '*',
  '*',
  1984-04-24T22:18:07.960000Z,
  2020-04-04T14:50:24.000000Z),
 ('TA',
  'Y18A',
  '*',
  '*',
  1984-04-24T22:18:07.960000Z,
  2020-04-04T14:50:24.000000Z),
 ('IU',
  'YAK',
  '*',
  '*',
  1984-04-24T22:18:07.960000Z,
  2020-04-04T14:50:24.000000Z),
 ('TA',
  'U46A',
  '*',
  '*',
  1984-04-24T22:18:07.960000Z,
  2020-04-04T14:50:24.000000Z),
 ('LD',
  'ODNJ',
  '*',
  '*',
  1984-04-24T22:18:0

In [179]:
net_reciever_pairs

[]

In [188]:
client = Client("IRIS")
bulk = net_reciever_pairs
inv = client.get_stations_bulk(bulk)

In [189]:
inv

Inventory created at 2022-03-22T12:22:50.102000Z
	Created by: IRIS WEB SERVICE: fdsnws-station | version: 1.1.48
		    
	Sending institution: IRIS-DMC (IRIS-DMC)
	Contains:
		Networks (145):
			2H, 4F, 5H, 7F, 9F, AC, AE, AF, AG, AK, AO, AT, AU, AV, AX, AZ, BC
			BE, BK, BL, C, C0, C1, CA, CB, CC, CH, CI, CM, CN, CO, CU, CW, CZ
			DK, EC, EI, EP, ET, G, GB, GE, GI, GM, GO, GR, GS, GT, HL, HT, HV
			HW, IC, II, IM, IN, IU, IW, JP, K5, KC, KG, KO, KP, KR, KS, KY, 
			KZ, LB, LD, LX, MB, MI, MN, N4, NC, NE, NI, NL, NM, NN, NP, NV, NY
			NZ, OK, ON, OO, PA, PB, PE, PM, PO, PR, PT, RM, RV, SB, SC, SE, 
			SF, SN, SV, TA, TJ, TM, TX, UO, US, UU, UW, VE, WI, WY, X2, X4, X8
			XD, XG, XI, XJ, XK, XT, XW, XY (2x), Y2, Y3, Y5, YA, YC, YF, YH, 
			YI, YJ, YM, YR, YV, YY, ZL, ZP, ZQ (2x), ZV, ZY
		Stations (2956):
			2H.ICEZ (Mt Erebus site ICE)
			2H.NAUS (Mt Erebus site NAUS)
			4F.VALP (American Legion 624)
			4F.VBB1 (VBB1)
			4F.VNLC (VNLC)
			4F.VSAB (VSAB)
			5H.ANNI (Yanceyville, VA)
			5H

In [159]:
stations_df_subset = stations_df[stations_df['station'].isin(metadata_stations)]

In [160]:
stations_df_subset

provider network_name station_name   latitude  longitude  \
0       BGR           GR          BUG    51.4412     7.2707   
0       BGR           GR          CLL   51.30803   13.00238   
0       BGR           GR          CLZ    51.8429    10.3741   
0       BGR           GR         GRA1  49.690776  11.220436   
0       BGR           GR         GRA2  49.654107  11.358088   
..      ...          ...          ...        ...        ...   
0       USP           ON        VAS01 -22.280081 -43.442624   
0       USP           II          AAK    42.6375    74.4942   
0       USP           II          KIV    43.9553    42.6863   
0       USP           II         PALK     7.2728    80.7022   
0       USP           II          ARU    56.4293    58.5615   

                 station_opened station_closed   station  
0   1991-09-01T00:00:00.000000Z           None    GR.BUG  
0   1993-04-01T00:00:00.000000Z           None    GR.CLL  
0   1991-02-01T00:00:00.000000Z           None    GR.CLZ  
0   1976-02-17T00:00:00.000000Z           None   GR.GRA1  
0   1976-02-17T00:00:00.000000Z           None   GR.GRA2  
..                          ...            ...       ...  
0   2011-09-14T00:00:00.000000Z           None  ON.VAS01  
0   1990-10-12T00:00:00.000000Z           None    II.AAK  
0   1994-02-01T00:00:00.000000Z           None    II.KIV  
0   2000-08-22T00:00:00.000000Z           None   II.PALK  
0   2009-09-28T00:00:00.000000Z           None    II.ARU  

[1919 rows x 8 columns]

In [149]:
stations_df['station_closed'].tolist()

[None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,
 None,

In [127]:
#stations_df = stations_df.append(station_row,ignore_index=False)

TypeError: Can only append a Series if ignore_index=True or if the Series has a name

In [139]:

#to_frame().T
row_frame = station_row.to_frame().T
row_frame.columns = stations_df.columns

ee = pd.concat([stations_df,row_frame],names = list(stations_df.columns), join = 'outer')

#stations_df.append(station_row.to_frame().T,ignore_index=True,)

In [140]:
ee

provider network_name station_name latitude longitude  \
0      USP           II          ERM   42.015  143.1572   

                station_opened station_closed  
0  1990-05-21T00:00:00.000000Z           None

In [82]:
stations_df

Empty DataFrame
Columns: [provider, network_name, station_name, latitude, longitude, station_opened, station_closed]
Index: []

In [58]:
list(stations_df.columns)

['provider',
 'network_name',
 'station_name',
 'latitude',
 'longitude',
 'station_opened',
 'station_closed']

In [33]:
len(stations_df.columns)

7

In [34]:
station_row

0    0
1    1
2    2
3    3
4    4
5    5
6    6
7    7
dtype: int64

In [74]:
stations_df = pd.concat([stations_df, station_row.T], axis = 0, ignore_index=False, names = list(stations_df.columns))
#,names =  list(stations_df.columns))

In [75]:
stations_df

0
0               Empty DataFrame
Columns: [provider, network_na...
1               provider          0
network_name      1
statio...
provider                                                        0
network_name                                                    1
station_name                                                    2
latitude                                                        3
longitude                                                       4
station_opened                                                  5
station_closed                                                  6

In [ ]:
row_frame = pd.Series([key,s.code,e.code,e.latitude,e.longitude,e.creation_date,e.termination_date]).to_frame().T                
#row_frame = station_row.to_frame().T
row_frame.columns = stations_df.columns

stations_df = pd.concat([stations_df,row_frame],names = list(stations_df.columns), join = 'outer')

In [167]:
#for s in stations.get_contents()["stations"]:
#stations_per_provider = {}
#providers = list(URL_MAPPINGS.keys())
#faulty_providers = []
#providers.remove('BGR')
faulty_providers_copy = faulty_providers.copy()
#faulty_providers.remove('EMSC')
#faulty_providers.remove('GFZ')

#faulty_providers.remove('GFZ')
for key in faulty_providers:
    #client = Client(key)
    #try:
    print(key)
    client = Client(key)
    stations_per_provider[key] = []
#client = Client("IRIS")

    stations = client.get_stations(starttime  = first_event,
    endtime=last_event,level='station')
    for s in stations:
        for e in s:
            row_frame = pd.Series([key,s.code,e.code,e.latitude,e.longitude,e.creation_date,e.termination_date]).to_frame().T                
            #row_frame = station_row.to_frame().T
            row_frame.columns = stations_df.columns

            stations_df = pd.concat([stations_df,row_frame],names = list(stations_df.columns), join = 'outer')
        #print()
            #try:
                #stations_per_provider[key].append(s.code + '.' + e.code)
            #except:
            #    pass
    #except:
    #    print(key)
        #faulty_providers.append(key)
        #pass
    #    raise ValueError
    #station_coords = {}
    #for s in stations:
    #    for e in s:
            #print()
    #        stations_per_provider[key].append(s.code + '.' + e.code)
            #station_coords[s.code + '.' + e.code] = (e.latitude,e.longitude)
            #print(s.code + '.' + e.code)
            #print(e.code)

    

IRIS


ValueError: value 336.68668 out of bounds (-90, 90)

In [163]:
faulty_providers_copy

NameError: name 'faulty_providers_copy' is not defined

In [162]:
faulty_providers

['EMSC', 'GFZ', 'IRIS', 'ISC', 'USGS']

In [ ]:
station_coords = {}
for s in stations:
    for e in s:
        #print()
        station_coords[s.code + '.' + e.code] = (e.latitude,e.longitude)
        #print(s.code + '.' + e.code)
        #print(e.code)


In [ ]:
df_iris_subset = df_test[df_test['station'].isin(station_coords.keys())]

In [ ]:
df_iris_subset

network_code receiver_code receiver_type  \
trace_name                                                            
AAK.II_20180115000700_NO             II           AAK            BH   
AAK.II_20180115001236_NO             II           AAK            BH   
AAK.II_20180115002306_NO             II           AAK            BH   
AAK.II_20180115003500_NO             II           AAK            BH   
AAK.II_20180115005518_NO             II           AAK            BH   
...                                 ...           ...           ...   
SNZO.IU_20161114015613_EV            IU          SNZO            BH   
SNZO.IU_20161115013448_EV            IU          SNZO            BH   
SNZO.IU_20161119204857_EV            IU          SNZO            BH   
SNZO.IU_20170204102921_EV            IU          SNZO            BH   
VAS01.ON_20141014200138_EV           ON         VAS01            HH   

                            receiver_latitude  receiver_longitude  \
trace_name                                                          
AAK.II_20180115000700_NO            42.637500           74.494200   
AAK.II_20180115001236_NO            42.637500           74.494200   
AAK.II_20180115002306_NO            42.637500           74.494200   
AAK.II_20180115003500_NO            42.637500           74.494200   
AAK.II_20180115005518_NO            42.637500           74.494200   
...                                       ...                 ...   
SNZO.IU_20161114015613_EV          -41.308700          174.704300   
SNZO.IU_20161115013448_EV          -41.308700          174.704300   
SNZO.IU_20161119204857_EV          -41.308700          174.704300   
SNZO.IU_20170204102921_EV          -41.308700          174.704300   
VAS01.ON_20141014200138_EV         -22.280081          -43.442624   

                            receiver_elevation_m  p_arrival_sample  \
trace_name                                                           
AAK.II_20180115000700_NO                  1633.1               NaN   
AAK.II_20180115001236_NO                  1633.1               NaN   
AAK.II_20180115002306_NO                  1633.1               NaN   
AAK.II_20180115003500_NO                  1633.1               NaN   
AAK.II_20180115005518_NO                  1633.1               NaN   
...                                          ...               ...   
SNZO.IU_20161114015613_EV                  120.0            1000.0   
SNZO.IU_20161115013448_EV                  120.0             700.0   
SNZO.IU_20161119204857_EV                  120.0             700.0   
SNZO.IU_20170204102921_EV                  120.0             800.0   
VAS01.ON_20141014200138_EV                 402.0             894.4   

                              p_status  p_weight  p_travel_sec  ...  \
trace_name                                                      ...   
AAK.II_20180115000700_NO           NaN       NaN           NaN  ...   
AAK.II_20180115001236_NO           NaN       NaN           NaN  ...   
AAK.II_20180115002306_NO           NaN       NaN           NaN  ...   
AAK.II_20180115003500_NO           NaN       NaN           NaN  ...   
AAK.II_20180115005518_NO           NaN       NaN           NaN  ...   
...                                ...       ...           ...  ...   
SNZO.IU_20161114015613_EV       manual      0.92         16.67  ...   
SNZO.IU_20161115013448_EV       manual      0.60         11.57  ...   
SNZO.IU_20161119204857_EV       manual      0.70         10.98  ...   
SNZO.IU_20170204102921_EV       manual      0.50         20.48  ...   
VAS01.ON_20141014200138_EV  autopicker      0.80          9.56  ...   

                            source_magnitude_author  \
trace_name                                            
AAK.II_20180115000700_NO                        NaN   
AAK.II_20180115001236_NO                        NaN   
AAK.II_20180115002306_NO                        NaN   
AAK.II_20180115003500_NO                        NaN   
AAK.II_20180115005518_NO                

In [ ]:
len(station_coords)

353

In [ ]:
stations

In [ ]:
len(stations['channels'])

2298

In [ ]:
coords

[]

In [ ]:
stations

['BL.APOB (FAZENDA SAO GERONIMO - APORE - GO)',
 'BL.APOB2 (FAZENDA SAO JOAO - APORE - GO)',
 'BL.AQDA (AQUIDAUANA - MS)',
 'BL.AQDB (Aquidauana, MS)',
 'BL.ARCA (Aracuai, MG)',
 'BL.AREB (AREADO - MG)',
 'BL.BAMB (BAMBUI - MG)',
 'BL.BB15B (FAZENDA CINTIA - BB - BEBEDOURO - SP)',
 'BL.BB16B (SITIO JOVAL - BB - BEBEDOURO - SP)',
 'BL.BB19B (Bebedouro 19, SP)',
 'BL.BB22B (Bebedouro 22, SP)',
 'BL.BEB (BELEM - PA)',
 'BL.BEB1 (SITIO STO ANTONIO - BEBEDOURO - SP)',
 'BL.BEB10 (FAZENDA SANTA ZULMIRA - BEBEDOURO - SP)',
 'BL.BEB11 (ESTANCIA DOURADA - ANDES - SP)',
 'BL.BEB12 (ESTANCIA MONIDENI - BEBEDOURO - SP)',
 'BL.BEB13 (FAZENDA RIO TURVO - BEBEDOURO - SP)',
 'BL.BEB14 (SITIO PALMEIRAS - BEBEDOURO - SP)',
 'BL.BEB15 (FAZENDA CINTIA  BEBEDOURO - SP)',
 'BL.BEB16 (SITIO JOVAL -  BEBEDOURO - SP)',
 'BL.BEB17 (SITIO SAO LEOPOLDO - BEBEDOURO - SP)',
 'BL.BEB18 (SITIO SAO PEDRO - BEBEDOURO - SP)',
 'BL.BEB19 (FAZENDA SANTA ZULMIRA - BEBEDOURO - SP)',
 'BL.BEB2 (FAZENDA SAO LUIZ -  BEBEDOURO 

In [ ]:
df_test_closest['network_code']

trace_name
KAN08.GS_20150408005359_EV    GS
Name: network_code, dtype: object

In [ ]:

client = Client("IRIS")
inventory = client.get_waveforms(network=df_test_closest['network_code'][0],
                                station=df_test_closest['receiver_code'][0],
                                starttime=UTCDateTime(df_test_closest['trace_start_time'][0].isoformat()),
                                endtime=UTCDateTime(df_test_closest['trace_start_time'][0].isoformat()) + 60,
                                location= "*",
                                channel =  "*",
                          attach_response=True)
                                #loc="*", 
                                #channel="*",
                                #level="response")  


In [ ]:
earliest_trace['trace_start_time'][0].isoformat()

'1993-01-10T04:38:49.790000'

In [ ]:
clients_dict = {}
for key in sorted(URL_MAPPINGS.keys()):
    clients_dict[key] = Client(key)

In [ ]:
clients_dict.keys()

dict_keys(['BGR', 'EMSC', 'ETH', 'GEONET', 'GFZ', 'ICGC', 'INGV', 'IPGP', 'IRIS', 'ISC', 'KNMI', 'KOERI', 'LMU', 'NCEDC', 'NIEP', 'NOA', 'ODC', 'ORFEUS', 'RASPISHAKE', 'RESIF', 'SCEDC', 'TEXNET', 'UIB-NORSAR', 'USGS', 'USP'])

In [ ]:

for key in sorted(URL_MAPPINGS.keys()):
    client = Client(key)
#for key,client in clients_dict.items():
    try:
        inventory = client.get_waveforms(network=earliest_trace['network_code'][0],
                                    station=earliest_trace['receiver_code'][0],
                                    starttime=UTCDateTime(earliest_trace['trace_start_time'][0].isoformat()),
                                    endtime=UTCDateTime(earliest_trace['trace_start_time'][0].isoformat()) + 60,
                                    location= "*",
                                    channel =  "*",
                            attach_response=True)

        print(key)
    except:
        pass
        #print(key)

NCEDC


In [ ]:
inventory

21 Trace(s) in Stream:

BK.MHC..BHE | 1993-01-10T04:38:49.828000Z - 1993-01-10T04:39:49.778000Z | 20.0 Hz, 1200 samples
...
(19 other traces)
...
BK.MHC..VHZ | 1993-01-10T04:38:56.286000Z - 1993-01-10T04:39:46.286000Z | 0.1 Hz, 6 samples

[Use "print(Stream.__str__(extended=True))" to print all Traces]

In [ ]:

for key in sorted(URL_MAPPINGS.keys()):
    client = Client(key)
#for key,client in clients_dict.items():
    try:
        inventory = client.get_waveforms(network=earliest_trace['network_code'][0],
                                    station=earliest_trace['receiver_code'][0],
                                    starttime=UTCDateTime(earliest_trace['trace_start_time'][0].isoformat()),
                                    endtime=UTCDateTime(earliest_trace['trace_start_time'][0].isoformat()) + 60,
                                    location= "*",
                                    channel =  "*",
                            attach_response=True)

        print(key)
    except:
        pass
        #print(key)

NCEDC


In [ ]:
inventory

4 Trace(s) in Stream:
NC.CAL.02.EHZ | 1984-04-24T22:18:07.960000Z - 1984-04-24T22:19:07.950000Z | 100.0 Hz, 6000 samples
NC.CAL.02.ELE | 1984-04-24T22:18:07.960000Z - 1984-04-24T22:19:07.950000Z | 100.0 Hz, 6000 samples
NC.CAL.02.ELN | 1984-04-24T22:18:07.960000Z - 1984-04-24T22:19:07.950000Z | 100.0 Hz, 6000 samples
NC.CAL.02.ELZ | 1984-04-24T22:18:07.960000Z - 1984-04-24T22:19:07.950000Z | 100.0 Hz, 6000 samples

In [ ]:
earliest_trace

client = Client("IRIS")
inventory = client.get_waveforms(network=earliest_trace['network_code'][0],
                                station=earliest_trace['receiver_code'][0],
                                starttime=UTCDateTime(earliest_trace['trace_start_time'][0].isoformat()),
                                endtime=UTCDateTime(earliest_trace['trace_start_time'][0].isoformat()) + 60,
                                location= "*",
                                channel =  "*",
                          attach_response=True)

FDSNNoDataException: No data available for request.
Detailed response of server:



In [ ]:
wave_resampled = inventory.resample(8000)
wave_resampled = wave_resampled.normalize()
one_sample = np.array(wave_resampled[0])
#wave_norm_x =  one_sample/one_sample.std()
#sf.write('assets/new_sound_extraction_test_other.wav',wave_norm_x,8000)
sf.write('assets/new_sound_extraction_test_obspy_normalize.wav',one_sample,8000)


In [ ]:
import librosa
import soundfile as sf
#Extract the proper seismic instrument
#index: 0: north-south, 1: east-west, 2:vertical
def create_seismic_sound(x, index = 0):
    #x1 = x[:,index]
    x1=np.array(x)
    norm_x = x1 = x1/x1.std()
    #norm_x_resampled = librosa.resample(norm_x, 100, 8000)
    sf.write('assets/new_sound_extraction_test.wav',norm_x_resampled,8000)

    return norm_x_resampled

In [ ]:
waveform_normalize = inventory.normalize()

In [ ]:
np.array(waveform_normalize[0]).shape

(12001,)

In [ ]:
waveform_normalize_resampled = librosa.resample(np.array(waveform_normalize[0]), 200, 8000)
sf.write('assets/new_sound_extraction_test.wav',waveform_normalize_resampled,8000)

<ipython-input-65-f6bdc02d7f2b>:1: FutureWarning: Pass orig_sr=200, target_sr=8000 as keyword args. From version 0.10 passing these as positional arguments will result in an error
  waveform_normalize_resampled = librosa.resample(np.array(waveform_normalize[0]), 200, 8000)


In [ ]:
create_seismic_sound(inventory[0])

<ipython-input-49-b8532cfac02a>:9: FutureWarning: Pass orig_sr=100, target_sr=8000 as keyword args. From version 0.10 passing these as positional arguments will result in an error
  norm_x_resampled = librosa.resample(norm_x, 100, 8000)


array([-0.04705812, -0.04743128, -0.04779273, ...,  0.02808716,
        0.02323079,  0.01848785])

In [ ]:
df_test_closest

network_code receiver_code receiver_type  \
trace_name                                                            
KAN08.GS_20150408005359_EV           GS         KAN08            HH   

                            receiver_latitude  receiver_longitude  \
trace_name                                                          
KAN08.GS_20150408005359_EV           37.22672          -97.970947   

                            receiver_elevation_m  p_arrival_sample p_status  \
trace_name                                                                    
KAN08.GS_20150408005359_EV                 375.0             400.0   manual   

                            p_weight  p_travel_sec  ...  \
trace_name                                          ...   
KAN08.GS_20150408005359_EV      0.63          0.78  ...   

                            source_magnitude_type source_magnitude_author  \
trace_name                                                                  
KAN08.GS_20150408005359_EV                     ml                    ISMP   

                            source_mechanism_strike_dip_rake  \
trace_name                                                     
KAN08.GS_20150408005359_EV                              None   

                           source_distance_deg source_distance_km  \
trace_name                                                          
KAN08.GS_20150408005359_EV              0.0004               0.04   

                           back_azimuth_deg  \
trace_name                                    
KAN08.GS_20150408005359_EV            166.8   

                                                           snr_db  \
trace_name                                                          
KAN08.GS_20150408005359_EV  [54.40000153 51.5        48.40000153]   

                            coda_end_sample        trace_start_time  \
trace_name                                                            
KAN08.GS_20150408005359_EV        [[1052.]] 2015-04-08 00:54:00.800   

                              trace_category  
trace_name                                    
KAN08.GS_20150408005359_EV  earthquake_local  

[1 rows x 34 columns]